### DISCUSSION TITLE
Get list of changed pages in onPostBuild

### DISCUSSION THREAD

**Discussion by gburgett** [gatsbyjs/gatsby#30498] [Help]
Hello,

I'm using Gatsby v3 to do incremental rebuilds of my site, powered by Contentful.  I would like to use the `onPostBuild` Node API to send out notifications when pages have changed.  Is there some way to get a list of all the new pages, updated pages, and deleted pages after each build?

**Comment by vladar** [comment]
Hey @gburgett 👋 

Currently, the only way to get this list is to use an undocumented `--write-to-file` option of gatsby CLI. The reason why it is not documented is that it will only contain pages that changed during the build. But build may produce other artifacts as well, e.g. images. Those artifacts won't be listed in this file.

Here is a comment about this option:

```
log-pages and write-to-file were added specifically to experimental GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES feature
in gatsby@2. They are useful, but not very applicable (specifically `--write-to-file`) as generic approach, as it only
list pages without other artifacts, so it's useful in very narrow scope. Because we don't have alternative right now
those toggles are kept for users that rely on them, but we won't promote them and will keep them "hidden".
```

So you can use it on your own risk. Also, unfortunately it is not available in `onPostBuild` as it is written after this callback is executed. But it is probably something we want to change so a PR is welcome!

Here is a related piece of code: https://github.com/gatsbyjs/gatsby/blob/bfce53344e4fac3c268065d1cbee4379baa959b1/packages/gatsby/src/commands/build.ts#L221-L285

**Reply by gburgett** [reply]
Thank you! 

I see that `onPostBuild` has access to the `getState()` function.  Is there a way to calculate this list of changes from the state during `onPostBuild`?
